delegation_tool.py

```diff
@@ -0,0 +1,387 @@
+        # Check to make sure user wants to update; show list of changes
+        for x in self.changes_summary:
+            print('\t'+x)
+            time.sleep(0.25)
+        LOGGER.log(LogLevel.WARN, "Push all to database? (y/n) ")
+        resp = input("")
+        if resp.strip() in ['y', 'Y']:
+            try:
+                self.survey_mongo_collection.update_one({'_id': self.survey_id}, {'$set': {'_id': self.survey_id, 'survey': self.survey}})
+            except:
+                # strip pictures out of json
+                for page in self.survey['pages']:
+                    for el in page['elements']:
+                        for pat in el.get('patients', []):
+                            if 'imgUrl' in pat and len(pat['imgUrl']) > 100:
+                                # see if imgurl is already in image database
+                                res = self.image_mongo_collection.find({'url': pat['imgUrl']})
+                                found = False
+                                for doc in res:
+                                    pat['imgUrl'] = doc['_id']
+                                    found = True
+                                    break
+                                if not found:
+                                    res = self.image_mongo_collection.insert_one({'url': pat['imgUrl']})
+                                    pat['imgUrl'] = res.inserted_id
+
+                # need to update one page at a time or else it gets too big
+                for k in self.survey:
+                    if k != 'pages':
+                        self.survey_mongo_collection.update_one({'_id': self.survey_id}, {'$set': {f'survey.{k}': self.survey[k]}})
+                for p in self.survey['pages']:
+                    res = self.survey_mongo_collection.find({'_id': self.survey_id, "survey.pages": {"$elemMatch": {"name": p['name']}}})
+                    doc_found = False
+                    for _ in res:
+                        doc_found = True
+                        break
+                    if doc_found:
+                        res = self.survey_mongo_collection.update_one({'_id': self.survey_id}, 
+                                            {'$set': {'survey.pages.$[elem]': p}}, array_filters=[{"elem.name": p['name']}], upsert=True)
+                    else:
+                        # push page 
+                        res = self.survey_mongo_collection.update_one({'_id': self.survey_id}, {'$push': {'survey.pages': p}}, upsert=True)
+            LOGGER.log(LogLevel.CRITICAL_INFO, "Survey successfully updated in mongo!")
+            self.changes_summary = []
+        else:
+            LOGGER.log(LogLevel.CRITICAL_INFO, "Okay, not updating survey.")
+
+
+    def import_page_from_json(self, json_dest, page_name, index=None):
+        '''
+        Move a page with page_name from a json file to the survey 
+        '''
+        self.cur_import_file = json_dest
+        f = open(json_dest, 'r')
```

Comment:
I forgot to mention-- both here and the other `open` call, for this to work on Windows, you need to add `encoding="utf-8"` to the call.  Derek and I ran into this in `_0_0_8_add_textbased_alignment.py`.

Reply:
That is good to know, thanks! I'll fix that too